exp/api/v1alpha1/azureasomanagedcluster_types.go

```diff
@@ -31,6 +32,11 @@ const (
 // AzureASOManagedClusterSpec defines the desired state of AzureASOManagedCluster.
 type AzureASOManagedClusterSpec struct {
 	AzureASOManagedClusterTemplateResourceSpec `json:",inline"`
+
+	// ControlPlaneEndpoint is the location of the API server within the control plane. CAPZ manages this field
+	// and it should not be set by the user. It fulfills Cluster API's cluster infrastructure provider contract.
+	//+optional
```

Comment:
A comment like this would be clarifying:

// Because this field is programmatically set by CAPZ after resource creation, we define it as +optional in the API schema to permit resource admission.